### ISSUE TITLE
Mug/Integration Setup: Config flow could not be loaded: 500 Internal Server Error Server got itself in trouble

### ISSUE THREAD

**Issue by brad07x** [sopelj/hass-ember-mug-component#27]
* Home Assistant Version (eg. 2023.2): 2023.2.5
* Ember Mug Component Version (eg. 0.5.0): 0.5.0-beta-6 and 0.5.0-beta-9; 0.5.0-beta-5 is last working version.

### Description

Attempting to pair/configure mug with HA following some changes to my Bluetooth adapter that were previously causing issues (added extension cable due to concerns over USB3 interference).

### What I Did

Deleted integration for previously paired mug with connection issues prior to USB/Bluetooth changes. 

Attempted to re-pair and reconfigure from discovered devices, and received the `Config flow could not be loaded: 500 Internal Server Error Server got itself in trouble` error when configuring the discovered integration for the mug.

The error is accompanied by an error in the HA log related to `helpers/data_entry_flow.py:44` including a traceback that I have included below.

Using HACS, I re-downloaded each 0.5.0 beta version 5-9 trying each and found that 0.5.0-beta-5 works as anticipated for the integration/configure screen, but I receive the error for 0.5.0-beta-6 and onward (thru the most recent 0.5.0-beta-9 release).

### Logs / Errors
```
Logger: aiohttp.server
Source: helpers/data_entry_flow.py:44
First occurred: 3:28:42 PM (1 occurrences)
Last logged: 3:28:42 PM
Error handling request

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/aiohttp/web_protocol.py", line 435, in _handle_request
    resp = await request_handler(request)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/web_middlewares.py", line 117, in impl
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 60, in security_filter_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 100, in forwarded_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 28, in request_context_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 80, in ban_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 236, in auth_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/view.py", line 145, in handle
    result = await result
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 171, in get
    return await super().get(request, flow_id)
  File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 100, in get
    result = self._prepare_result_json(result)
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 184, in _prepare_result_json
    return _prepare_config_flow_result_json(result, super()._prepare_result_json)
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 121, in _prepare_config_flow_result_json
    return prepare_result_json(result)
  File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 44, in _prepare_result_json
    data["data_schema"] = voluptuous_serialize.convert(
  File "/usr/local/lib/python3.10/site-packages/voluptuous_serialize/__init__.py", line 40, in convert
    pval = convert(value, custom_serializer=custom_serializer)
  File "/usr/local/lib/python3.10/site-packages/voluptuous_serialize/__init__.py", line 113, in convert
    if issubclass(schema, Enum):
TypeError: issubclass() arg 1 must be a class
```

**Comment by sopelj** [comment]
Oh, sorry about that. I added a new option and it caused an issue. It should be fixed. I created a tenth beta.

If you are removing and re-adding a mug and it is not automatically re-discovered you might need to go into `bluetoothctl` and remove the previous one with `remove``then it shoudl pop up.

**Comment by brad07x** [comment]
Thanks @sopelj !

0.5.0-beta-10 resolves the error I described, but the extra option appeared as a checkbox with no label in the config flow:

![Blank Checkbox Image](https://i.imgur.com/TBhbqKu.png[/img)

I did complete a restart of HA following redownload of 0.5.0-beta-10, but not sure if I reloaded the devices page before starting the config flow (if that would make a difference).

That said, the config flow did indeed work and my mug paired!

On my end the changes I made to the adapter setup seem to have improved connectivity as well. I had bluetoothctl open in a terminal, but didn't have to remove the mug first before entering & pairing. Your new read/write functionality also worked - I was able to update the name of the mug.

Thanks again for all your work on this project! If you feel like it, you should add a 'buymeacoffee' link or similar to the project page.

**Comment by sopelj** [comment]
Awesome! I'm glad it works now!

You don't need to manually remove the device in order to re-add it, you just need to remove it to get the notification for autodiscovery. (However, sometime you can still get it after reboot)

That's really strange that you don't have the text. I wonder if the old one is still cached or something. I have the text for the new entry on mine. What language is your Home Assistant set to? Regardless it should fall back to the default keys.

![screenshot-2023-02-19-19-48-32](https://user-images.githubusercontent.com/5285941/219986291-e7a35b2a-bd31-4349-92bc-700b11cac152.png)

I appreciate the suggestion, I might, but I'm just glad it can be of use. 

Also, unrelated, but I see you have yours set to fahrenheit, so I was wondering, are your readings correct? [A few have mentionned issues with fahrenheit](https://github.com/sopelj/hass-ember-mug-component/issues/15#issuecomment-1415058314). It should be fixed, but I haven't heard back about it yet from them yet.

**Comment by brad07x** [comment]
I would wager that the label for that option was my fault for not reloading the HA UI in the browser tab.

On your question about temp readings in Fahrenheit, it does look like mine are working. 72F/matching room temp when idle and 160-170F when adding hot water. Reasonably close. I'll see if it stabilizes at the 135F setpoint of my mug in the reading and then add a note to the issue you linked.

Thanks again!

**Comment by sopelj** [comment]
Oh that could do it!

Awesome! Thank you very much!

The non beta release should be out in a couple days if no other issues crop up.

Don't hesitate if there are any other issues or suggestions.

**Comment by sopelj** [comment]
Ok, fixed in 0.5.0. Don't hesitate if there is anything else.
